kbase-extension/static/kbase/js/common/spec.js

```diff
@@ -101,11 +101,51 @@ define(['bluebird', 'util/util', 'common/sdk', 'widgets/appWidgets2/validators/r
             return Promise.props(validationMap);
         }
 
+        /**
```

Comment:
IMO, the changes here are pretty straightforward. Validate stuff in a set using the Resolver.